src/resources/UsageAnalytics/Read/ReadServiceCommon.ts

```diff
@@ -0,0 +1,132 @@
+export module ParamParts {
```

Comment:
The reasoning was to "hide" the partial interfaces I'm using to build specific Params interfaces in a submodule, so they don't show up on the main export. I initially started having them all named with a `ParamsPart` suffix in the name, which triggered me to just put them in a submodule. I thought it also clarified that they were used to compose actual params interfaces. But I don't mind reverting to having the parts as "global" module types with suffixes in the names.

Reply:
What do you mean by 'global' module and 'main' export?

If you don't export the interface in the entrypoint, then they're not exposed to the user fo the library and all's well no?

Reply:
> What do you mean by 'global' module and 'main' export?

By "global module" I mean the root object that is exported by the library, and all types that are on there. Main export was a different way of referring to the same thing.

I don't see any point for users of this library to see any of the "ParamParts" interfaces on the import. In other words, I see no reason they should be able to do:
```typescript
import {OrganizationParamPart, TimeRangeParamPart} from ''@coveo/platform-client';
```

I know they can still get it through `ParamParts`, which is exported, but with this all the partial interfaces don't show up at the top level.

> If you don't export the interface in the entrypoint, then they're not exposed to the user of the library and all's well no?

Maybe my knowledge of TypeScript hasn't aged well, but you shouldn't have interfaces referenced on any public signature (class/interface via implements/extends, method via parameter/return type) that isn't exported? Otherwise you're referring to a type that is unknown (hidden inside your bundled module). The [API extractor page](https://api-extractor.com/) still refers to this as missing exports.